crypto/ec/ec_key.c

```diff
@@ -350,6 +351,76 @@ static int ec_generate_key(EC_KEY *eckey, int pairwise_test)
     return ok;
 }
 
+#ifndef FIPS_MODULE
+int ossl_ec_public_from_private(EC_KEY *eckey, BIGNUM *priv_key, BN_CTX *ctx)
+{
+    int ok = 0;
+    EC_POINT *pub_key = NULL;
+    const EC_GROUP *group = eckey->group;
+
+    if (eckey->pub_key == NULL) {
+        pub_key = EC_POINT_new(group);
+        if (pub_key == NULL)
+            goto err;
+    } else {
+        pub_key = eckey->pub_key;
+    }
+
+    /* pub_key = priv_key * G (where G is a point on the curve) */
+    if (!EC_POINT_mul(group, pub_key, priv_key, NULL, NULL, ctx))
+        goto err;
+
+    eckey->pub_key = pub_key;
+    ok = 1;
+err:
+    if (!ok) {
+        if (eckey->pub_key != NULL)
+            EC_POINT_set_to_infinity(group, eckey->pub_key);
+    }
+    return ok;
+}
```

Comment:
ossl_ec_key_simple_generate_public_key() is basically the point_multiply line..
Not sure that warrants melding the two? I could call this code from the generate but that doesnt make things simpler.

Reply:
I guess my concern was why does that `keygenpub` function pointer exist in EC_METHOD at all? Are there circumstances where you would need a different implementation? I can't think what those circumstances might be, and AFAICT there don't seem to be any alternative implementations anyway. If anyone *does* introduce such an implementation then this could introduce a subtle bug here because it wouldn't be used.

It also seems odd to have that function, and then introduce a second essentially identical implementation.

Reply:
I have got rid of the function as it was only needed internally.. Removing the function does to some extent mean the code I needed just shifted somewhere else, so this doesnt achieve very much. I was trying to make the code look like the other generate function. I call the function used by keygenpub to run the couple of lines of code to do a multiply now. Note that the generate() function doesnt do that, which is what I was trying to copy. NOTE: The keygenpub() is called in one place by a d2i method.